tests/test-extensions/heisenberg-extension/src/main/java/org/mule/test/heisenberg/extension/HeisenbergScopes.java

```diff
@@ -47,6 +47,7 @@ public void executeAnything(Chain chain, CompletionCallback<Void, Void> cb) {
     chain.process(cb::success, (t, e) -> cb.error(t));
   }
 
+  @MediaType(value = TEXT_PLAIN, strict = false)
```

Comment:
You add this because a validation started to fail, right? 
So you need to detect this case and use -> `org.mule.runtime.module.extension.internal.loader.validation.ModelValidationUtils#isCompiletime` to ignore if the error occurs in a non compiletime scenario.

Reply:
Review all modified tests and check if they run correctly if you ignore the validation on `compiletime`

Reply:
I'll comment here for tracking purpose

Ignoring the validation if not in _compiletime_,  all tests of the test case `MediaTypeModelValidatorTestCase` that don't expect an exception to be thrown are green.

On test cases `JavaDeclarationDelegateTestCase` and `DynamicMetadataDeclarationEnricherTestCase` the metadataType is obtain from the instrocpectionUtils itself insteed of harcoded it.

Without the added annotation on `HeisenbergScopes` the only test case that fails is `ExtensionActivatorTestCase, which is not relevant on not compile time.

Reply:
I compiled the java modules with 4.2.0-SNAPSHOT and all test passed as well.